diff --git a/mojoal.c b/mojoal.c
index 1cb1594..7ec266f 100644
--- a/mojoal.c
+++ b/mojoal.c
@@ -11,9 +11,11 @@
 #include <math.h>
 #include <float.h>
 
-#ifdef _WIN32
-  #define AL_API __declspec(dllexport)
-  #define ALC_API __declspec(dllexport)
+#ifndef AL_LIBTYPE_STATIC
+  #ifdef _WIN32
+    #define AL_API __declspec(dllexport)
+    #define ALC_API __declspec(dllexport)
+  #endif
 #endif
 
 #ifndef M_PI
