diff --git a/cmake/sdlchecks.cmake b/cmake/sdlchecks.cmake
index 8dcb39265..59d66c473 100644
--- a/cmake/sdlchecks.cmake
+++ b/cmake/sdlchecks.cmake
@@ -409,6 +409,7 @@ macro(CheckX11)
         /opt/X11/include
     )
 
+    set(CMAKE_REQUIRED_INCLUDES "${CMAKE_REQUIRED_INCLUDES}")
     if(X_INCLUDEDIR)
       list(APPEND EXTRA_CFLAGS "-I${X_INCLUDEDIR}")
       list(APPEND CMAKE_REQUIRED_INCLUDES "${X_INCLUDEDIR}")
