diff --git a/cmake/sdlchecks.cmake b/cmake/sdlchecks.cmake
index c4aec174e..2b8502a68 100644
--- a/cmake/sdlchecks.cmake
+++ b/cmake/sdlchecks.cmake
@@ -438,6 +438,7 @@ macro(CheckX11)
         /opt/X11/include
     )
 
+    set(CMAKE_REQUIRED_INCLUDES "${CMAKE_REQUIRED_INCLUDES}")
     if(X_INCLUDEDIR)
       list(APPEND EXTRA_CFLAGS "-I${X_INCLUDEDIR}")
       list(APPEND CMAKE_REQUIRED_INCLUDES ${X_INCLUDEDIR})
