diff --git a/src/core/android/SDL_android.c b/src/core/android/SDL_android.c
index 34dfdedf..bb188d01 100644
--- a/src/core/android/SDL_android.c
+++ b/src/core/android/SDL_android.c
@@ -1056,19 +1056,19 @@ JNIEXPORT void JNICALL SDL_JAVA_INTERFACE(onNativeSurfaceChanged)(JNIEnv *env, j
 {
     SDL_LockMutex(Android_ActivityMutex);

-#if SDL_VIDEO_OPENGL_EGL
-    if (Android_Window) {
-        SDL_VideoDevice *_this = SDL_GetVideoDevice();
-        SDL_WindowData *data = (SDL_WindowData *)Android_Window->driverdata;
-
-        /* If the surface has been previously destroyed by onNativeSurfaceDestroyed, recreate it here */
-        if (data->egl_surface == EGL_NO_SURFACE) {
-            data->egl_surface = SDL_EGL_CreateSurface(_this, (NativeWindowType)data->native_window);
-        }
-
-        /* GL Context handling is done in the event loop because this function is run from the Java thread */
-    }
-#endif
+//#if SDL_VIDEO_OPENGL_EGL
+//    if (Android_Window) {
+//        SDL_VideoDevice *_this = SDL_GetVideoDevice();
+//        SDL_WindowData *data = (SDL_WindowData *)Android_Window->driverdata;
+//
+//        /* If the surface has been previously destroyed by onNativeSurfaceDestroyed, recreate it here */
+//        if (data->egl_surface == EGL_NO_SURFACE) {
+//            data->egl_surface = SDL_EGL_CreateSurface(_this, (NativeWindowType)data->native_window);
+//        }
+//
+//        /* GL Context handling is done in the event loop because this function is run from the Java thread */
+//    }
+//#endif

     SDL_UnlockMutex(Android_ActivityMutex);
 }
